packages/contracts/src/chugsplash/deploy.ts

```diff
@@ -0,0 +1,45 @@
+/* Imports: External */
+import { injectL2Context } from '@eth-optimism/core-utils'
+import hre from 'hardhat'
+import { ethers } from 'ethers'
+
+/* Imports: Internal */
+import { getContractFactory } from '..'
+import { makeActionBundleFromConfig } from './hardhat-tools'
+
+const DEPLOYER_PRIVATE_KEY = process.env.DEPLOYER_PRIVATE_KEY
+const L2_NODE_URL = process.env.L2_NODE_URL
+const CHUGSPLASH_DEPLOYER_ADDRESS = process.env.CHUGSPLASH_DEPLOYER_ADDRESS
+const UPGRADE_CONFIG_PATH = process.env.UPGRADE_CONFIG_PATH
+
+const deploy = async () => {
+  const l2Provider = injectL2Context(
+    new ethers.providers.JsonRpcProvider(L2_NODE_URL)
+  )
+  const signer = new ethers.Wallet(DEPLOYER_PRIVATE_KEY, l2Provider)
```

Comment:
Can you add support for a mnemonic + hdpath and ledger?

Reply:
will base it off https://github.com/ethereum-optimism/secret-service/blob/a6d0bfb407101803c49d4f7ad43721be749a4344/src/services/withdrawal-submitter/run.ts

Reply:
we'll need to add `@ethersproject/hardware-wallets` as a dependency to contracts, is that what we want / imagine the deployment process will look like? not sure how much bloat this repo should have @tynes @smartcontracts, maybe we can have this as a future todo as we finalize how l2 deployments happen

Reply:
It used to be a dependency of the contracts until `bin/deploy` was rewritten. I don't think its going to bloat the repo too much and if it does then we can make it a `peerDependency`. We should strive towards never using company funds without using a ledger unless the key must be hot